registry/registry.go

```diff
@@ -49,7 +48,7 @@ func (r *registrar) NamePrefix() string {
 // Find looks at its internal index to find a registration that matches the
 // entity instance provided. Return an error when not found.
 func (r *registrar) Find(entity dosa.DomainObject) (*dosa.RegisteredEntity, error) {
-	name := reflect.TypeOf(entity).Name()
+	name := reflect.TypeOf(entity).Elem().Name()
```

Comment:
+1 to comment by @junchaowu -- this needs some kind of test added to make sure it doesn't break again. It's clear this code is not getting covered.